backend/src/services/federation.ts

```diff
@@ -0,0 +1,45 @@
+import { BasePeerConnector } from '../connectors/peer/base.js'
+import getPeerConnector from '../connectors/peer/index.js'
+import peer, { getPeerIds } from '../connectors/peer/index.js'
+import { FederationState, PeerConfigStatus } from '../types/types.js'
+import config from '../utils/config.js'
+import { ConfigurationError, InternalError } from '../utils/error.js'
+
+function ensureFederationNotGloballyDisabled() {
+  if (FederationState.DISABLED == config.federation.state) {
+    throw InternalError('Federation is globally disabled')
+  }
+}
+
+function ensureFederationNotDisabled(connector: BasePeerConnector) {
+  if (FederationState.DISABLED == connector.getConfiguredState()) {
+    throw ConfigurationError(`Federation is currently disabled; will not fetch peer ${connector.getId()} status`)
+  }
+}
+
+export async function getAllPeerStatus(): Promise<Map<string, PeerConfigStatus>> {
+  ensureFederationNotGloballyDisabled()
+  const peerStatus = new Map<string, PeerConfigStatus>()
+  const promises: Array<Promise<any>> = []
```

Comment:
Is there a reason that this is set to `any`? I could be wrong but my understanding is that it's an array of whatever `getPeerStatus` returns, which I think is `SystemStatus`?